packages/botkit/src/core.ts

```diff
@@ -689,6 +694,35 @@ export class Botkit {
         }
     }
 
+    /**
+     * Converts an incoming message into BotkitMessage activity
+     * @param message The incoing message from adapter
+     * @return {BotkitMessage} the newly created BotkitMessage activity
+     */
+    private incomingMessageToBotkitMessage(message: any): BotkitMessage {
+        const { channelData } = message;
+        const activity = {
+            // start with all the fields that were in the original incoming payload. NOTE: this is a shallow copy, is that a problem?
+            ...message,
+            // timestamp: new Date(),
+            type: message.type === 'message' ? ActivityTypes.Message : ActivityTypes.Event,
+            // normalize the user, text and channel info
+            event: channelData.type || channelData.botkitEventType || message.type,
+            text: message.text,
+            user: message.from.id,
+            channel: message.conversation.id,
+            value: message.value,
+            incoming_messge: message,
```

Comment:
Should this not be `incoming_message`?

Reply:
Yes good catch I will probably not merge this piece anyways since it is sort of out of scope of the main change.